### DISCUSSION TITLE
App shows a blank screen on other devices

### DISCUSSION THREAD

**Discussion by Virock** [unoplatform/uno#16182] [Q&A]
Hi everyone.
I finished writing my app using Uno Platform. It works perfectly well on my PC. I packaged the Release version of the app into a .msix package. I installed this package and it works perfectly well on my PC.
Unfortunately, when I copy the file over to my other PC, it installs but when I launch the app, I get a blank screen.
I actually learned about the blank screen from Microsoft. I submitted the app to the Microsoft Partner program and they responded with this message: 
They see a blank screen.

How do I fix this?

**Comment by jeromelaban** [comment]
@Virock there may be various reasons for this, but in general, this means that there's an exception happening somewhere in the app. Best is to log as much as possible for your app to determine which exceptions may be happening. Uno Platform templates providing [logging capabilities](https://platform.uno/docs/articles/logging.html) that you can use for your WinAppSDK app.

**Reply by Virock** [reply]
Thank you so much. I found the bug. I was trying to create a folder in a place where I did not have write permissions.

**Reply by Virock** [reply]
Quick question. Is there a hook I can use to tell when the file system has finished initializing?
I tried writing this ```StorageFolder localFolder = Windows.Storage.ApplicationData.Current.LocalFolder;``` in Application.OnLaunched but it causes the app to crash. I saw this article https://platform.uno/docs/articles/features/file-management.html that explained that the initialization of the filesystem is asynchronous so it might not be available. Is there a way to know when it's ready?

**Reply by jeromelaban** [reply]
@Virock  The end of this [document section](https://platform.uno/docs/articles/features/file-management.html#webassembly-file-system) explains how do to it. Any asynchronous function from `StorageFile`/`StorageFolder` will wait for the initialization to be done.
